### DISCUSSION TITLE
Ability to click a button (menu, etc.) in the partially visible window without scrolling the view to the window

### DISCUSSION THREAD

**Discussion by hipnoizz** [paperwm/PaperWM#941] [Ideas]
Hello. :)

A window may be partially visible. When I interact with the window - click some button and so on so it becomes focused - Paper WM will 'scroll' the view to show show that window fully. I wonder if there is an option to interact with a window without it being scrolled into the view.

An example use case: I have an active file manager window on the right half of the screen. I have a partially visible Firefox window on the left. I see that Firefox finished downloading the file, so I want to click on the 'Downloads' button on its toolbar and select 'Show in folder' to show the folder in the file manager. Clicking on the 'Downloads' button in Firefox will make the view to scroll to the Firefox and potentially make the file manager window disappear from the view. So then I need to scroll right (e.g. by `Super+Right`) to view the file manager window.

This is just an example - I may want to copy some visible content from Firefox and paste it in another window that I'm focused on and so on.

Also if I right-click on a partially visible window to bring up a context menu the behaviour is a bit not user-friendly. In Firefox after its window is scrolled into the view the context menu will disappear. In other application the context menu will stay visible, but it won't be under the mouse cursor any more.

I thought if maybe clicking with Ctrl/Alt/Shift/Super could work that way, but a) it is not very user-friendly either, and b) given application may have some action bound to to click+modifier combination.

Personally I think it might work for me if there was an option to disable automatic scrolling into the view when the window is clicked. I could still scroll to the window using shortcuts or mouse-scrolling on the Window Position Bar.

I'm not sure if this is something achievable. A window (application) may be activated in many ways - by clicking using the mouse, maybe using some shortcuts etc. etc.

**Comment by jtaala** [comment]
Hey @hipnoizz, this is a pretty good idea.  It's actually fairly trivial to implement something like this - essentially it just controlling this block of code:

https://github.com/paperwm/PaperWM/blob/2c3e76a63951fa2f278e89ffc9cacfa59e5fe994/tiling.js#L4312-L4320

E.g. if you comment that out it won't scroll to window view.

I tried just disabling that - didn't feel right.  I then disabled it just when a window was focused (i.e. when clicking with mouse etc.) and left scrolling active when switching to a window via shortcut or opening a new window etc.  Felt a little better.
